cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -726,8 +730,9 @@ private BatchPhaseSpec getPhaseSpec(String programName, PipelinePhase phase) {
     // create a local dataset to store alerts. At the end of the phase, the dataset will be scanned and alerts
     // published.
     for (StageSpec alertPublisherInfo : phase.getStagesOfType(AlertPublisher.PLUGIN_TYPE)) {
-      String stageName = alertPublisherInfo.getName();
-      AlertPublisherSink alertPublisherSink = new AlertPublisherSink(stageName, null);
+      String datasetName = UUID.randomUUID().toString();
```

Comment:
I think it's a little more understandable if we name them alerts1, alerts2, etc. Then at least we know the datasets are for alert publishing. I've seen a couple times the local datasets not get removed for some reason, and it's nicer if they have a name that isn't UUID1.UUID2